src/services/redisSessionStorage.ts

```diff
@@ -2,52 +2,128 @@
  * This service uses the Redis client to store and retrieve session information.
  */
 
-import { Either, left } from "fp-ts/lib/Either";
+import { Either, isLeft, left, right } from "fp-ts/lib/Either";
 import * as redis from "redis";
 import { User } from "../types/user";
 import { extractUserFromJson } from "../types/user";
 import { ISessionStorage } from "./iSessionStorage";
+import TokenService from "./tokenService";
+
+const SESSION_NOT_FOUND_MESSAGE = "Session not found";
+
+export interface ISessionState {
+  readonly expired: boolean;
+  readonly expireAt: string;
+  readonly newToken?: string;
+  readonly user?: User;
+}
 
 export default class RedisSessionStorage implements ISessionStorage {
   constructor(
     private readonly redisClient: redis.RedisClient,
-    private readonly tokenDuration: number
+    private readonly tokenDuration: number,
```

Comment:
`tokenDuration` should make explicit the unit of measure (e.g. `tokenDurationSecs` or `tokenDurationMillis`)